demo-kotlin/app/src/gms/java/com/example/placesdemo/CurrentPlaceTestActivity.kt

```diff
@@ -60,89 +70,260 @@ class CurrentPlaceTestActivity : AppCompatActivity() {
             Place.Field.OPENING_HOURS,
             Place.Field.PHONE_NUMBER,
             Place.Field.UTC_OFFSET,
-            Place.Field.WEBSITE_URI)
+            Place.Field.WEBSITE_URI
+        )
         fieldSelector = FieldSelector(
             findViewById(R.id.use_custom_fields),
             findViewById(R.id.custom_fields_list),
             savedInstanceState,
-            placeFields)
+            placeFields,
+        )
         responseView = findViewById(R.id.response)
         setLoading(false)
 
         // Set listeners for programmatic Find Current Place
-        findViewById<Button>(R.id.find_current_place_button).setOnClickListener {
-            findCurrentPlace()
-        }
+        findViewById<Button>(R.id.find_current_place_button)
+            .setOnClickListener { findCurrentPlace() }
     }
 
     override fun onSaveInstanceState(bundle: Bundle) {
         super.onSaveInstanceState(bundle)
         fieldSelector.onSaveInstanceState(bundle)
     }
 
+    @SuppressLint("MissingPermission")
+    override fun onRequestPermissionsResult(
+        requestCode: Int,
+        permissions: Array<String>,
+        grantResults: IntArray
+    ) {
+        if (requestCode != PERMISSION_REQUEST_CODE) {
+            super.onRequestPermissionsResult(
+                requestCode,
+                permissions,
+                grantResults
+            )
+            return
+        }
+
+        if (isLocationPermissionGranted(permissions, grantResults)) {
+            findCurrentPlaceWithPermissions()
+        }
+    }
+
+    private fun isLocationPermissionGranted(
+        permissions: Array<String>,
+        grantResults: IntArray
+    ): Boolean =
+        permissions.toList().zip(grantResults.toList())
+            .firstOrNull { (permission, grantResult) ->
+                grantResult == PackageManager.PERMISSION_GRANTED && (ACCESS_FINE_LOCATION == permission || ACCESS_COARSE_LOCATION == permission)
```

Comment:
Is it idiomatic to check equality with the constant first and the variable second? I am used to `permission == ACCESS_FINE_LOCATION`

Reply:
It is idiomatic to do it in the style that you are used to. Will change.

I was thinking in Java and had written this as `ACCESS_FINE_LOCATION.equals(permission)` but later refactored it since `==` is the same as `equals` in Kotlin. This style is common in Java where you have your constant first before the variable that you are checking equality of to prevent NPEs. But, this is Kotlin so it's less of an issue.